### DISCUSSION TITLE
Installing CadQuery on CloudLinux 7

### DISCUSSION THREAD

**Discussion by hootsh1337** [CadQuery/cadquery#1611] [Q&A] [answered]
Good evening,

Is there anyway I could install CadQuery on CloudLinux 7 OS ?

There is a server where we're trying to build a Django app, and connect it to a server, to upload some CadQuery apps to be used online.

Due to an issue with the server, I can only use a certain specific python "venv" pip virtual environment.

All trials ends with the error: `Can't find the module: OCP`

Things I tried so far to install OCP:
1- Installing using pip directly - can't
2- Installing using .whl file from pip - doesn't work, tried many different versions
3- Building the package .whl directly - freezes when using  `pywrap all ocp.toml`
4- Installing conda then using conda/mamba inside the pip environment - doesn't work

At this point I don't think it is supported, is it?

**Comment by adam-urbanczyk** [comment]
What is the exact failure mode of (2)? Maybe the maintainers of the wheel could help with more data. If all fails, you could pull the generated sources from the OCP release and compile yourself. There is not need (it is even not advised) to run pywrap.

**Reply by hootsh1337** [reply]
It just says:
ERROR: Could not find a version that satisfies the requirement (name of whl file)
ERROR: No matching distribution found for (name of whl file)

I tried almost all .whl files and all python versions at this point.

As for building, you mean use "cmake" directly? I'm not sure what happens there to be honest, I just did as instructed in the repo.

**Reply by hootsh1337** [reply]
One more thing, I don't know if it's relevant. But the server admin sent me a msg saying it's a missing "Apache" file.

He said he'll look into it, but at this point, do you think there is anything I could say, to point him in the right direction?

**Reply by Jopie01** [reply, marked as answer]
> It just says: ERROR: Could not find a version that satisfies the requirement (name of whl file) ERROR: No matching distribution found for (name of whl file)
> 
> I tried almost all .whl files and all python versions at this point.

I think your Python version is too old. I have a CentOS 7 VM running and the default Python version is 2.7.5. I also have Python 3.6 installed from the repositories, but I think that version is also already too old for this.

I see a few possibilities here:
1. If possible upgrade to a newer version of CloudLinux ( 8 or 9)
2. Build one of the newest Python versions from source
3. Look into Podman (www.podman.io) or Docker to get a container and install everything in that container. Preferred way is to use Podman so you don't need root access to run the container


> One more thing, I don't know if it's relevant. But the server admin sent me a msg saying it's a missing "Apache" file.

 That has nothing to do with this, Apache is a webserver.

**Reply by hootsh1337** [reply]
I see, alright I'll try to work on all of these options.
Thank you very much for the guidance.

**Comment by Jopie01** [comment]
Just some questions came up:
- I don't know CloudLinux, but it seems a CentOS 7 continuity?
- Do you have SSH connection to the server?
- What Python version are you using? 

I have successfully installed CadQuery many times now on both RockyLinux 8, RockyLinux 9 and Debian 12. What I did was:
- install the most recent version of Python supplied by the repositories
- create a new 'venv' with `python3.11 -m venv --system-site-packages cadquery-env`
- activate the 'venv'
- install CadQuery from git with `pip install git+https://github.com/CadQuery/cadquery.git`

The last command installs everything. Maybe you need some extra packages to be installed because of OpenGL etc.

**Reply by hootsh1337** [reply]
That is informative, thank you. Glad to know it should work, in theory at least.

I do have a SSH to the server. I also tried different versions of python. But my SSH is a bit restricted so maybe that's an issue.

The server provides the option to use several pre-made venvs with diff python versions for our Django python app, so i tried all of them. 

It is a shared server, so there are some things "restricted" in my SSH access. I did contact the admin and he's looking into it, but I don't have much info to give him.

Last msg I received from him is that it's a missing "Apache" file and he'll look into it.

**Comment by jmwright** [comment]
Older pip versions have trouble with the newer wheels. You can try to run `pip install --upgrade pip` to see if it solves the issue.

**Reply by hootsh1337** [reply]
Thank you for the reply.

I did try to update pip, but it made no difference.
